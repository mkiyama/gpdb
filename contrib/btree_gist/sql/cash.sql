-- money check

-- In PostgreSQL, this test uses WITH OIDS, but in GPDB, OIDs are not unique
-- across segments, so use a 'serial' column instead.
CREATE TABLE moneytmp (a money, oid serial);

\copy moneytmp (a) from 'data/cash.data'

SET enable_seqscan=on;

SELECT count(*) FROM moneytmp WHERE a <  '22649.64';

SELECT count(*) FROM moneytmp WHERE a <= '22649.64';

SELECT count(*) FROM moneytmp WHERE a  = '22649.64';

SELECT count(*) FROM moneytmp WHERE a >= '22649.64';

SELECT count(*) FROM moneytmp WHERE a >  '22649.64';

CREATE INDEX moneyidx ON moneytmp USING gist ( a );

SET enable_seqscan=off;

SELECT count(*) FROM moneytmp WHERE a <  '22649.64'::money;

SELECT count(*) FROM moneytmp WHERE a <= '22649.64'::money;

SELECT count(*) FROM moneytmp WHERE a  = '22649.64'::money;

SELECT count(*) FROM moneytmp WHERE a >= '22649.64'::money;

SELECT count(*) FROM moneytmp WHERE a >  '22649.64'::money;
