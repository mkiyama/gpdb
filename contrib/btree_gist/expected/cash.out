-- money check
-- In PostgreSQL, this test uses WITH OIDS, but in GPDB, OIDs are not unique
-- across segments, so use a 'serial' column instead.
CREATE TABLE moneytmp (a money, oid serial);
NOTICE:  CREATE TABLE will create implicit sequence "moneytmp_oid_seq" for serial column "moneytmp.oid"
\copy moneytmp (a) from 'data/cash.data'
SET enable_seqscan=on;
SELECT count(*) FROM moneytmp WHERE a <  '22649.64';
 count 
-------
   289
(1 row)

SELECT count(*) FROM moneytmp WHERE a <= '22649.64';
 count 
-------
   290
(1 row)

SELECT count(*) FROM moneytmp WHERE a  = '22649.64';
 count 
-------
     1
(1 row)

SELECT count(*) FROM moneytmp WHERE a >= '22649.64';
 count 
-------
   254
(1 row)

SELECT count(*) FROM moneytmp WHERE a >  '22649.64';
 count 
-------
   253
(1 row)

CREATE INDEX moneyidx ON moneytmp USING gist ( a );
SET enable_seqscan=off;
SELECT count(*) FROM moneytmp WHERE a <  '22649.64'::money;
 count 
-------
   289
(1 row)

SELECT count(*) FROM moneytmp WHERE a <= '22649.64'::money;
 count 
-------
   290
(1 row)

SELECT count(*) FROM moneytmp WHERE a  = '22649.64'::money;
 count 
-------
     1
(1 row)

SELECT count(*) FROM moneytmp WHERE a >= '22649.64'::money;
 count 
-------
   254
(1 row)

SELECT count(*) FROM moneytmp WHERE a >  '22649.64'::money;
 count 
-------
   253
(1 row)

